semgrep-core/src/engine/Match_rules.ml

```diff
@@ -111,15 +112,30 @@ type env = {
   pattern_matches : id_to_match_results;
   (* used by metavariable-pattern to recursively call evaluate_formula *)
   file : Common.filename;
-  rule_id : string;
+  rule_id : R.rule_id;
   xlang : R.xlang;
   equivalences : Equivalence.equivalences;
+  (* problems found during evaluation, one day these may be caught earlier by
+   * the meta-checker *)
```

Comment:
yep, exactly, for each of those errors now surfaced to the user we should improve the metachecker to detect them too
statically.